report/metrics.go

```diff
@@ -49,10 +53,45 @@ type Sample struct {
 	Value     float64   `json:"value"`
 }
 
-// MakeMetric makes a new Metric.
-// TODO: Specialized version adding the first sample to avoid generating garbage?
-func MakeMetric() Metric {
-	return Metric{}
+// MakeSingletonMetric makes a metric with a single value
+func MakeSingletonMetric(t time.Time, v float64) Metric {
```

Comment:
It's not singleton in the OOP sense but in the mathematical sense https://en.wikipedia.org/wiki/Singleton_(mathematics) (I have borrowed this from Haskell).

Any other naming suggestions?